test/integration/addons_test.go

```diff
@@ -190,7 +190,57 @@ func testServicesList(t *testing.T) {
 		t.Fatalf(err.Error())
 	}
 }
+func testRegistry(t *testing.T) {
+	t.Parallel()
+	minikubeRunner := NewMinikubeRunner(t)
+	kubectlRunner := util.NewKubectlRunner(t)
+	minikubeRunner.RunCommand("addons enable registry", true)
+	t.Log("wait for registry to come up")
+
+	if err := util.WaitForDockerRegistryRunning(t); err != nil {
+		t.Fatalf("waiting for registry to be up: %v", err)
+	}
+
+	// Check access from outside the cluster on port 5000, validing connectivity via registry-proxy
+	checkExternalAccess := func() error {
+		t.Log("checking registry access from outside cluster")
+		expectedStr := "200"
+		runCmd := fmt.Sprintf("curl -sS -o /dev/null -w '%%{http_code}' http://127.0.0.1:5000")
```

Comment:
Please use the retryable http instead of curl, not every system has curl and it is not rertiable 

please change to be like this one
https://github.com/kubernetes/minikube/blob/master/test/integration/proxy_test.go#L153

or this one https://github.com/kubernetes/minikube/blob/e236c966a74d7952b9940e2f7bac1ee1fe38239a/test/integration/version_upgrade_test.go#L38